### PR TITLE
Add a BUILD file in src/core

### PR DESCRIPTION
**Pull request by yashykt**
This needs to be cherry-picked

### CODE REVIEW

File: src/core/BUILD [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+package(
+    default_visibility = ["//visibility:public"],
+    features = [
+        "layering_check",
+        "-parse_headers",
+    ],
+)
+
+grpc_cc_library(
```
Comment (ctiller) [inline src/core/BUILD:37]: Can we do `event_engine_common` before getting too much further... it's got a dependency on `include/` that I think is going to inform the rest of the decisions we might make on this.

Comment (yashykt) [reply src/core/BUILD:37]: Thanks for the suggestion!

**Review by veblush — APPROVED** [review]

File: src/core/BUILD [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+load(
+    "//bazel:grpc_build_system.bzl",
+    "grpc_cc_library",
+)
+
+licenses(["reciprocal"])
+
+package(
+    default_visibility = ["//visibility:public"],
```
Comment (ctiller) [inline src/core/BUILD:23]: I think this is inadvisable.

File: src/core/BUILD [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    features = [
+        "layering_check",
+    ],
+)
+
+exports_files(glob(["**"]))
```
Comment (ctiller) [inline src/core/BUILD:29]: Can you say what this is for?
What visibility should it have?
Is there a chance to remove it?

File: tools/distrib/fix_build_deps.py [hunk 4]
```diff
@@ -498,7 +507,10 @@ def best(self, scorer):
             choices = new_choices
 
         best = None
-        final_scorer = lambda x: (total_avoidness(x), scorer(x), total_score(x))
+
+        def final_scorer(x):
```
Comment (ctiller) [inline tools/distrib/fix_build_deps.py:511]: Why can't this be a lambda?

File: tools/distrib/fix_build_deps.py [hunk 5]
```diff
@@ -296,15 +296,22 @@ def grpc_cc_library(name,
         proto_hdr = '%s%s' % ((parsing_path + '/' if parsing_path else ''),
                               proto.replace('.proto', '.pb.h'))
         skip_headers[name].add(proto_hdr)
+
+    # Convert the source or header target to a relative path.
+    def get_filename(name, parsing_path):
```
Comment (ctiller) [inline tools/distrib/fix_build_deps.py:301]: functions should be at global scope, not inlined

**Review by ctiller — COMMENTED** [review]
Some questions about visibility and plans for such before this is submitted.

File: tools/distrib/fix_build_deps.py (hunk 4 above)
Comment (yashykt) [reply tools/distrib/fix_build_deps.py:511]: blame yapf :P

File: tools/distrib/fix_build_deps.py (hunk 5 above)
Comment (yashykt) [reply tools/distrib/fix_build_deps.py]: Done

File: src/core/BUILD (hunk 3 above)
Comment (yashykt) [reply src/core/BUILD]: Thanks!

File: src/core/BUILD (hunk 2 above)
Comment (yashykt) [reply src/core/BUILD]: Good point, Thanks!

File: tools/distrib/fix_build_deps.py (hunk 4 above)
Comment (yashykt) [reply tools/distrib/fix_build_deps.py]: It was actually my vscode formatter. Reverted.

**Review by ctiller — APPROVED** [review]

**Pull request merged**